File: src/p0078/java/solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        int n = nums.length, pn = 1 << n;
        List<List<Integer>> result = new ArrayList<>();
        for (int pm = 0; pm < pn; pm++) {
            ArrayList<Integer> temp = new ArrayList<>();
            for (int k = 0; k < n; k++) {
                if (((1 << k) & pm) != 0) {
                    temp.add(nums[k]);
                }
            }
            result.add(temp);
        }
        return result;
    }
}

File: src/p0078/cpp/solution.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<vector<int>> subsets(vector<int> &nums) {
        auto n = nums.size(), pn = (1UL << n);
        vector<vector<int>> result(pn);
        for (auto pm = 0; pm < pn; pm++) {
            for (auto k = 0; k < n; k++) {
                if ((1 << k) & pm) {
                    result[pm].push_back(nums[k]);
                }
            }
        }
        return result;
    }
};
